zipline/_protocol.pyx

```diff
@@ -45,6 +45,11 @@ cdef class BarData:
         try:
             view = self._views[asset]
         except KeyError:
+            try:
+                asset = self.data_portal.env.asset_finder.retrieve_asset(asset)
```

Comment:
In that case it is just gets itself back, so a noop-y type operation.

On Wed, Jan 6, 2016, 5:31 PM Jean Bredeche <notifications@github.com> wrote:

> In zipline/_protocol.pyx
> <https://github.com/quantopian/zipline/pull/943#discussion_r49018961>:
>
> > @@ -45,6 +45,11 @@ cdef class BarData:
> >          try:
> >              view = self._views[asset]
> >          except KeyError:
> > +            try:
> > +                asset = self.data_portal.env.asset_finder.retrieve_asset(asset)
>
> what if asset is already an Asset? is that possible?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/quantopian/zipline/pull/943/files#r49018961>.
>